lib/value-pairs/cmdline.c

```diff
@@ -340,8 +340,38 @@ vp_cmdline_parse_rekey_shift (const gchar *option_name, const gchar *value,
       return FALSE;
     }
 
-  value_pairs_transform_set_add_func
-  (vpts, value_pairs_new_transform_shift (atoi (value)));
+  value_pairs_transform_set_add_func(vpts,
+                                     value_pairs_new_transform_shift (atoi (value)));
+  return TRUE;
+}
+
+static gboolean
+vp_cmdline_parse_rekey_shift_levels (const gchar *option_name, const gchar *value,
+                                     gpointer data, GError **error)
+{
+  gpointer *args = (gpointer *) data;
+  ValuePairsTransformSet *vpts = (ValuePairsTransformSet *) args[2];
+  gchar *key = (gchar *) args[3];
+  gchar *end = NULL;
+  gint number_to_shift = strtol(value, &end, 0);
+
+  if (number_to_shift <= 0 || *end != 0)
+    {
+      g_set_error (error, G_OPTION_ERROR, G_OPTION_ERROR_FAILED,
+                   "Error parsing value-pairs, argument to --shift-levels is not numeric or not a positive number");
+      return FALSE;
+    }
+
+  vpts = vp_cmdline_rekey_verify (key, vpts, data);
+  if (!vpts)
+    {
+      g_set_error (error, G_OPTION_ERROR, G_OPTION_ERROR_FAILED,
+                   "Error parsing value-pairs: --shift-levels used without --key or --rekey");
+      return FALSE;
+    }
+
+  value_pairs_transform_set_add_func(vpts,
+                                     value_pairs_new_transform_shift_levels (atoi (value)));
```

Comment:
Thanks for adding the validation. A few lines above we convert the `value` string to `integer` with `strol`, but in this line we redo the calculation with `atoi`. My question is, do we need to use different conversion function, or even would be better, can we reuse the already calculated integer here too?

Reply:
ugh, sorry. I certainly meant that. Thank you for catching this.